context/src/main/java/io/opentelemetry/context/ContextValue.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.context;
+
+/**
+ * A value that can be stored inside {@link Context}. Types will generally use this interface to
+ * allow storing themselves in {@link Context} without exposing a {@link ContextKey}.
+ */
+public interface ContextValue {
```

Comment:
Just thinking out loud about naming. 
What about 
"Propagatable"
"ContextAware"
"ContextPropagatable"
"Contextual"
"PropagationAware"
"PropagationSupport"
"ContextSupport"

Reply:
AutoContextStorable
ContextKeyAssociated
ImplicitlyContextKeyed
WithImplicitContextKey

Reply:
I like `ContextStorable`, not auto since a user still has to call `withValue` so not so automatic, and `store` is similar naming to the method name. `ImplicitContextKeyed` seems nice too for explaining the why if not necessarily the how. I don't know how to pick among any of the names here, what should I do? :)

Reply:
+1 for `ImplicitContextKeyed`

Reply:
How about just `ContextKeyed` ?